gazelle/pythonconfig/pythonconfig.go

```diff
@@ -134,12 +134,20 @@ type Configs map[string]*Config
 
 // ParentForPackage returns the parent Config for the given Bazel package.
 func (c *Configs) ParentForPackage(pkg string) *Config {
-	dir := path.Dir(pkg)
-	if dir == "." {
-		dir = ""
+	for {
+		dir := path.Dir(pkg)
+		if dir == "." {
+			dir = ""
+		}
+		parent := (map[string]*Config)(*c)[dir]
+		if parent != nil {
+			return parent
+		}
+		if dir == "" {
+			return nil
+		}
+		pkg = dir
```

Comment:
Could you add a unit test for this new logic? Just testing the `Configs` in isolation would be sufficient.

Right now it is a little bit hard to read and having tests would help understand what this has to do.

Reply:
I've added a few basic tests for this method.

There should probably be some additional full tests for `update_only` and mixing that with the different `python_generation_mode` modes but I'm not familiar enough with it to do in this initial PR.